qubesmanager/settings.py

```diff
@@ -619,12 +624,104 @@ def __apply_advanced_tab__(self):
         except qubesadmin.exc.QubesException as ex:
             msg.append(str(ex))
 
+        try:
+            if self.virt_mode.currentIndex() != self.virt_mode_idx:
+                self.vm.virt_mode = self.selected_virt_mode().lower()
+                # pylint: disable=attribute-defined-outside-init
+                self.anything_changed = True
+        except Exception as ex:  # pylint: disable=broad-except
+            msg.append(str(ex))
+
         return msg
 
     def boot_from_cdrom_button_pressed(self):
         self.save_and_apply()
         subprocess.check_call(['qubes-vm-boot-from-device', self.vm.name])
 
+    def selected_virt_mode(self):
+        return self.virt_mode_list[self.virt_mode.currentIndex()]
+
+    def virt_mode_changed(self, new_idx):  # pylint: disable=unused-argument
+        self.update_pv_warning()
+        self.update_pvh_dont_support_devs()
+        self.update_pvh_kernel_ver_warning()
+
+    def update_pv_warning(self):
+        if self.selected_virt_mode() == 'PV':
+            self.pv_warning.show()
+        else:
+            self.pv_warning.hide()
+
+    def update_virt_mode_list(self):
+        choices = ['HVM', 'PV']
+
+        if hasattr(self, 'dev_list'):
+            devs_attached = self.dev_list.selected_list.count() != 0
+        else:
+            devs_attached = bool(list(self.vm.devices['pci'].persistent()))
+
+        if devs_attached:
+            self.pvh_mode_hidden.show()
+        else:
+            choices.insert(0, 'PVH')
+            self.pvh_mode_hidden.hide()
+
+        if hasattr(self, 'virt_mode_list'):
+            old_mode = self.selected_virt_mode()
+            self.virt_mode.currentIndexChanged.disconnect()
+        else:
+            old_mode = None
+
+        self.virt_mode.clear()
+
+        # XXX: Hardcoded default value.
```

Comment:
`qubesadmin.QubesVM` now have `property_get_default()` method. When using it, add version dependency to spec file: `Requires: python3-qubesadmin >= 4.0.13`.
This is specifically relevant, because we've just changed the default to PVH...